kratos/python_scripts/analysis_stage.py

```diff
@@ -344,6 +349,30 @@ def __CreateListOfProcesses(self):
         self._list_of_output_processes = self._CreateProcesses("output_processes", order_processes_initialization)
         self._list_of_processes.extend(self._list_of_output_processes) # Adding the output processes to the regular processes
 
+    def _GetMapOfAdditionalHistoricalVariables(self):
+        """This function returns the list of (additional) historicsl variables
+        needed in the simulation
+        It can be overridden in derived classes
+        """
+        map_additional_vars = defaultdict(set)
+        self._GetLMapOfAdditionalVariablesFromProcesses("processes", map_additional_vars)
+        self._GetLMapOfAdditionalVariablesFromProcesses("output_processes", map_additional_vars)
+
+        return map_additional_vars
+
+    def _GetLMapOfAdditionalVariablesFromProcesses(self, parameter_name, map_additional_vars):
+        """This function extracts the required historical variables needed by the processes

```

Comment:
can you improve the docs please?
I.e. add sth like the vars for the corresponding ModelParts

Reply:
done!